client/layout/store-notices/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import PropTypes from 'prop-types';
+import { Notice } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+class StoreWarnings extends Component {
+	isVisibleMissingPaimentWarning() {
+		const { paymentMethods, queryString } = this.props;
+		const isPaymentPage = queryString.task
+			? queryString.task === 'payments'
+			: false;
+		return ! Object.keys( paymentMethods ).length && isPaymentPage;
+	}
+
+	render() {
+		const showMissingPaimentMethod = this.isVisibleMissingPaimentWarning();
+
+		return (
+			showMissingPaimentMethod && (
+				<Notice
+					className="woocommerce-store-warnings"
+					status="warning"
+					isDismissible={ false }
+				>
+					<p>
+						{ __(
+							'Customers cannot place orders at your store until you setup and enable a payment method.',
+							'woocommerce-admin'
+						) }
+					</p>
+				</Notice>
+			)
+		);
+	}
+}
+
+StoreWarnings.propTypes = {
+	/**
+	 * The url query string.
+	 */
+	queryString: PropTypes.object,
+};
+
+export default compose(
+	withSelect( ( select ) => {
+		const { getOption } = select( OPTIONS_STORE_NAME );
+
+		const optionNames = [
```

Comment:
I found that I had four options returned here, even after removing all of the payment provider plugins I had sitting around. I'm guessing that the settings option doesn't get removed when removing the plugin. Deleting the options made this work. You might want to see if there's a different way to check if a payment provider is installed - maybe using `getActivePlugins()` - https://github.com/woocommerce/woocommerce-admin/blob/main/client/homescreen/stats-overview/install-jetpack-cta.js#L165

Reply:
Hi @becdetat, thank you for your feedback.
Your idea sounds good to me.

Just to confirm:
if the user has selected one of the following payment methods:  `PayPal, PayFast, Karna Checkout, Stripe, and Square` we should verify if the plugins related to those methods are also activated. If they aren't activated we'll assume the user doesn't want those methods, even though they appear as active in our options. Did I understand correctly?

Thinking about this, I wonder if we should also update the state of the inactive payment gateways to not have inconsistencies with the information we have in the DB.

Reply:
Really good point @becdetat - that hardcoded list of options caused me to pause a bit when I saw it too.

What do you all think about possibly just calling out to the core WC payment gateways endpoint: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-payment-gateways

That might require a bit more work though since I don't think we have a data store setup for that yet, but maybe an [apiFetch](https://developer.wordpress.org/block-editor/packages/packages-api-fetch/) could work. 

Alternatively, we could consider this kind of an edge case for now - especially given the context of this warning being shown in the Setup Checklist.